sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -504,12 +506,12 @@ static int sof_audio_probe(struct platform_device *pdev)
 
 	if (soc_intel_is_byt() || soc_intel_is_cht()) {
 		is_legacy_cpu = 1;
-		dmic_num = 0;
+		dmic_be_num = 0;
 		hdmi_num = 0;
 		/* default quirk for legacy cpu */
 		sof_rt5682_quirk = SOF_RT5682_SSP_CODEC(2);
 	} else {
-		dmic_num = 1;
+		dmic_be_num = 2;
```

Comment:
Should dmic_be_num be defined by quirk?

Reply:
We haven't found that on SKL+ yet, @singalsu please correct me if I am wrong.

If we do have only one PDM on some platforms(not machines), we can set it here(just like set it to be 0 for legacy cpu) when found.